### ISSUE TITLE
Add `failing` tests feature

### ISSUE THREAD

**Issue by victordidenko** [lukeed/uvu#47]
Ava has [failing tests feature](https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#failing-tests):

```javascript
test.failing('demonstrate some bug', t => {
  t.fail(); // Test will count as passed
});
```

It behaves like normal test, just with opposite result — failed tests considered passed, i.e. they are expected to be failed.

It is useful when there is known issue, and I plan to fix it later — I can write test for that issue, and expect this test to fail _just yet_, until I fix issue, and mark test as usual _passing_ test then.

Maybe mark dots for such tests with different color in output.

**Comment by lukeed** [comment]
Hey, thank you for the suggestion.

However, `test.failing` never made any sense to me. It's using a test block as an assertion modifier -- what?? IMO, just write the opposite assertion you expected. It's made easier in `uvu` too, given that each assertion method has a `not` modifier.

Option 2 is to write the tests you expect to work, and then use `test.skip` to skip execution entirely. You should always work towards having 0 skipped items, and the summary view will help remind you that you're non-zero.

But as you say, a "passing" `test.failing` isn't passing at all.  And it's not failing. It's a weird in-between that shouldn't exist.. _and_ it's taking the role of assertions in the first place... _and_ you'll have to rewrite the entire test once the issue is fixed. If you're asserting an error state, then write assertions for the error state.

TLDR: Don't think it makes sense to have, and the current `uvu` tools allow you to achieve the same thing but  with more useful & consistent meaning.

Hope that makes sense, or at least sheds some light on my thinking.

Thank you though :)

**Comment by victordidenko** [comment]
I think you understand it wrong. This is not _opposite assertion_, this is _opposite test block_, it can contain many assertions.

> and you'll have to rewrite the entire test once the issue is fixed

No, and this is key point. You write test as _if an issue has been fixed already_, and test _should_ pass. But until you actually fix the issue — this test will fail, so you mark it as `failing`. After you fix the issue — you remove `failing` modifier and thats it, you are perfect, having the test for the fixed issue.
You don't change any assertions inside test, you just remove `failing` modifier, telling, that issue has been fixed.
This also is useful, if your QA can write test, and reference it in the bug report.

But I got your point, skiping test could do the trick, at least for me.

Thank you :)

**Comment by victordidenko** [comment]
Also, as far as I remember from Ava, failing tests are still prints errors in the output, like usual failed tests, they just don't fail your testing results. So, CI still works, but you see errors, and this is far more eye-catching, than small number of skipped tests.

**Comment by willstott101** [comment]
I try to explain the use case to the Jest guys here if it helps: https://github.com/facebook/jest/issues/10030

I love the additional context xfail tests give you in the output - vs inverted logic for every assertion...
